src/main/java/fr/flowarg/flowupdater/utils/builderapi/ModdedBuilder.java

```diff
@@ -0,0 +1,116 @@
+package fr.flowarg.flowupdater.utils.builderapi;
+
+import fr.flowarg.flowupdater.download.json.*;
+import fr.flowarg.flowupdater.utils.ModFileDeleter;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public abstract class ModdedBuilder<T, B extends ModdedBuilder<T, B>> implements IBuilder<T>
+{
+    protected final BuilderArgument<List<Mod>> modsArgument = new BuilderArgument<List<Mod>>("Mods", ArrayList::new).optional();
+    protected final BuilderArgument<List<CurseFileInfo>> curseModsArgument = new BuilderArgument<List<CurseFileInfo>>("CurseMods", ArrayList::new).optional();
+    protected final BuilderArgument<List<ModrinthVersionInfo>> modrinthModsArgument = new BuilderArgument<List<ModrinthVersionInfo>>("ModrinthMods", ArrayList::new).optional();
+    protected final BuilderArgument<ModFileDeleter> fileDeleterArgument = new BuilderArgument<>("ModFileDeleter", () -> new ModFileDeleter(false)).optional();
+    protected final BuilderArgument<CurseModPackInfo> curseModPackArgument = new BuilderArgument<CurseModPackInfo>("CurseModPack").optional();
+    protected final BuilderArgument<ModrinthModPackInfo> modrinthPackArgument = new BuilderArgument<ModrinthModPackInfo>("ModrinthModPack").optional();
+
+    protected abstract B getThis();
+
+    /**
+     * Append a mods list to the version.
+     * @param mods mods to append.
+     * @return the builder.
+     */
+    public B withMods(List<Mod> mods)
+    {
+        this.modsArgument.get().addAll(mods);
+        return getThis();
+    }
+
+    /**
+     * Append a mods list to the version.
```

Comment:
"Append a single mod or a mod array to the version." serait plus précis par rapport au comportement de la méthode ?

Reply:
C'est t'as javadoc, j'y ait rien touché, mais si tu veux